test/rbi/rewriters/replace_attributes_with_methods_test.rb

```diff
@@ -0,0 +1,292 @@
+      # We'll support both, until Sorbet starts to prefer one or the other.
+
+      rbi = Parser.parse_string(<<~RBI)
+        # Lorum ipsum...
+        sig { params(a: Integer).returns(Integer) }
+        attr_writer :a
+      RBI
+
+      rbi.replace_attributes_with_methods!
+
+      assert_equal(<<~RBI, rbi.string)
+        # Lorum ipsum...
+        sig { params(a: Integer).void }
+        def a=(a); end
+      RBI
+    end
+
+    def test_replaces_attr_accessor_with_getter_and_setter_methods
+      rbi = Parser.parse_string(<<~RBI)
+        # Lorum ipsum...
+        sig { returns(Integer) }
+        attr_accessor :a
+      RBI
+
+      rbi.replace_attributes_with_methods!
+
+      assert_equal(<<~RBI, rbi.string)
+        # Lorum ipsum...
+        sig { returns(Integer) }
+        def a; end
+
+        # Lorum ipsum...
+        sig { params(a: Integer).void }
+        def a=(a); end
+      RBI
+    end
+
+    ### Testing for multiple attributes defined in a single declaration
+
+    def test_replaces_multi_attr_reader_with_methods
+      rbi = Parser.parse_string(<<~RBI)
+        # Lorum ipsum...
+        sig { returns(Integer) }
+        attr_reader :a, :b, :c
+      RBI
+
+      rbi.replace_attributes_with_methods!
+
+      assert_equal(<<~RBI, rbi.string)
+        # Lorum ipsum...
+        sig { returns(Integer) }
+        def a; end
+
+        # Lorum ipsum...
+        sig { returns(Integer) }
+        def b; end
+
+        # Lorum ipsum...
+        sig { returns(Integer) }
+        def c; end
+      RBI
+    end
+
+    def test_replaces_multi_attr_writer_with_methods
+      rbi = Parser.parse_string(<<~RBI)
+        # Lorum ipsum...
+        sig { params(a: Integer).void }
+        attr_writer :a, :b, :c
+      RBI
+
+      rbi.replace_attributes_with_methods!
+
+      assert_equal(<<~RBI, rbi.string)
+        # Lorum ipsum...
+        sig { params(a: Integer).void }
+        def a=(a); end
+
+        # Lorum ipsum...
+        sig { params(b: Integer).void }
+        def b=(b); end
+
+        # Lorum ipsum...
+        sig { params(c: Integer).void }
+        def c=(c); end
+      RBI
+    end
+
+    def test_replaces_multi_attr_accessor_with_methods
+      rbi = Parser.parse_string(<<~RBI)
+        # Lorum ipsum...
+        sig { returns(Integer) }
+        attr_accessor :a, :b, :c
+      RBI
+
+      rbi.replace_attributes_with_methods!
+
+      assert_equal(<<~RBI, rbi.string)
+        # Lorum ipsum...
+        sig { returns(Integer) }
+        def a; end
+
+        # Lorum ipsum...
+        sig { returns(Integer) }
+        def b; end
+
+        # Lorum ipsum...
+        sig { returns(Integer) }
+        def c; end
+
+        # Lorum ipsum...
+        sig { params(a: Integer).void }
+        def a=(a); end
+
+        # Lorum ipsum...
+        sig { params(b: Integer).void }
+        def b=(b); end
+
+        # Lorum ipsum...
+        sig { params(c: Integer).void }
+        def c=(c); end
+      RBI
```

Comment:
On second thought, I think it might be better to group by property name, rather than by reader vs writer, i.e.:

```ruby
        sig { returns(Integer) }
        def a; end

        sig { params(a: Integer).void }
        def a=(a); end

        sig { returns(Integer) }
        def b; end

        sig { params(b: Integer).void }
        def b=(b); end

        sig { returns(Integer) }
        def c; end

        sig { params(c: Integer).void }
        def c=(c); end
```